### DISCUSSION TITLE
Unable to install Homebrew on M2 Mac with Sonoma 14.2.1

### DISCUSSION THREAD

**Discussion by hemanshukumar** [Homebrew/discussions#5048] [Getting started]
### Output of `brew config`

```shell
zsh: command not found: brew
```

### Output of `brew doctor`

```shell
zsh: command not found: brew
```

### Description of issue

I had installed Homebrew a few months ago, and it worked fine. Today it started giving errors while attempting to install a cask (the errors were while homebrew was auto-updating itself prior to installing the new cask). 

Long story short, I ended up uninstalling Homebrew completely, and now I cannot get it to reinstall.

Here is some of the output when I try to install Homebrew using the official script:

...
```
HEAD is now at dca9ff865 Merge pull request #16465 from MikeMcQuaid/user_working_directory_readable
fatal: cannot force update the branch 'master' checked out at '/opt/homebrew'
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:d783cbeb6e6ef0d71c0b442317b54554370decd6fac66bf2d4938c07a63f67be
######################################################################### 100.0%
Can't load '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' for module Digest::SHA: dlopen(/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle, 0x0001): tried: '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (no such file), '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')) at /System/Library/Perl/5.30/darwin-thread-multi-2level/DynaLoader.pm line 197.
 at /usr/bin/shasum5.30 line 25.
Compilation failed in require at /usr/bin/shasum5.30 line 25.
BEGIN failed--compilation aborted at /usr/bin/shasum5.30 line 25.
Error: Checksum mismatch.
Expected: d783cbeb6e6ef0d71c0b442317b54554370decd6fac66bf2d4938c07a63f67be
  Actual: 
 Archive: /Users/hemanshu/Library/Caches/Homebrew/portable-ruby-3.1.4.arm64_big_sur.bottle.tar.gz
To retry an incomplete download, remove the file above.
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
Failed during: /opt/homebrew/bin/brew update --force --quiet
```

**Comment by SMillerDev** [comment]
Did you retry removing the file?

**Comment by hemanshukumar** [comment]
Do you mean 
`/Users/hemanshu/Library/Caches/Homebrew/portable-ruby-3.1.4.arm64_big_sur.bottle.tar.gz`?

Once I uninstall Homebrew after a failed attempt, there is no such file or even folder on my machine.

I have also tried removing just that file and attempting the install again; it makes no difference to the outcome.

**Comment by hemanshukumar** [comment]
Update: I tried using the `Homebrew-4.2.3.pkg` file from [here](https://github.com/Homebrew/brew/releases/tag/4.2.3) to install Homebrew, instead of the terminal command. The installation succeeds. I also was able to run `/opt/homebrew/bin/brew shellenv` to put the output into my `~/.zprofile` file. However, running `brew doctor` or `brew config` still produces the same problems as before:

```
hemanshu@Tres-Leches ~ % brew config
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:d783cbeb6e6ef0d71c0b442317b54554370decd6fac66bf2d4938c07a63f67be
######################################################################### 100.0%
Can't load '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' for module Digest::SHA: dlopen(/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle, 0x0001): tried: '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (no such file), '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')) at /System/Library/Perl/5.30/darwin-thread-multi-2level/DynaLoader.pm line 197.
 at /usr/bin/shasum5.30 line 25.
Compilation failed in require at /usr/bin/shasum5.30 line 25.
BEGIN failed--compilation aborted at /usr/bin/shasum5.30 line 25.
Error: Checksum mismatch.
Expected: d783cbeb6e6ef0d71c0b442317b54554370decd6fac66bf2d4938c07a63f67be
  Actual: 
 Archive: /Users/hemanshu/Library/Caches/Homebrew/portable-ruby-3.1.4.arm64_big_sur.bottle.tar.gz
To retry an incomplete download, remove the file above.
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```

And again, deleting the `/Users/hemanshu/Library/Caches/Homebrew/portable-ruby-3.1.4.arm64_big_sur.bottle.tar.gz` file and attempting to `brew config` again does not help.

**Comment by Satarupa22-SD** [comment]
Hi, my bad I only looked at the system specifications and posted it here. My issue is resolved apparently I just needed some software update and a quick restart.

**Comment by DavidGzzMilan** [comment]
Hi, any comments? I just migrated from a MacBook x86_64 to arm64 (M1) and got the same issue when trying to get and use Homebrew. I also was able to install it with .pkg file but ended in the same situation:
```
➜  ~ brew config
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:d783cbeb6e6ef0d71c0b442317b54554370decd6fac66bf2d4938c07a63f67be
#=#=-  #       #                                                                                                                                                                                                                                               #=O#-     #        #                                                                                                                                                                                                                                           -#O=- #      #          #                                                                                                                                                                                             ######################################################################################################################################################################################################################################################### 100.0%
Can't load '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' for module Digest::SHA: dlopen(/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle, 0x0001): tried: '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (no such file), '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')) at /System/Library/Perl/5.30/darwin-thread-multi-2level/DynaLoader.pm line 197.
 at /usr/bin/shasum5.30 line 25.
Compilation failed in require at /usr/bin/shasum5.30 line 25.
BEGIN failed--compilation aborted at /usr/bin/shasum5.30 line 25.
Error: Checksum mismatch.
Expected: d783cbeb6e6ef0d71c0b442317b54554370decd6fac66bf2d4938c07a63f67be
  Actual:
 Archive: /Users/davidgonzalez/Library/Caches/Homebrew/portable-ruby-3.1.4.arm64_big_sur.bottle.tar.gz
To retry an incomplete download, remove the file above.
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```

**Reply by SMillerDev** [reply]
Your system Perl is failing to calculate shasums. Nothing Homebrew can help you with unfortunately.

**Comment by hemanshukumar** [comment]
Update: since I didn't get any resolution from here, I was hoping that updates from Apple's side might resolve it. No such luck. I have now updated to Sonoma 14.3, and the issue persists.

I should also note that, like @DavidGzzMilan, I too migrated from an Intel Mac a few months ago, and originally Homebrew was installed on that machine.

**Comment by delfireinoso** [comment]
I've installed Homebrew on a M2 and on a Intel Mac under Sonoma without any problem. Perhaps the issues are migrating the installation form an Intel mac to a ARM Mac

**Comment by hemanshukumar** [comment]
@delfireinoso that's been my (uneducated) guess so far too, especially given this bit of the error:
`'/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))`

However, I should mention that it doesn't seem to be something that always happens with such migrations. For one, we'd expect to see many more people with this problem then. For another, I myself migrated from an Intel iMac to one with the M3 chip with no problems at all. This problem has come up after migrating from an intel MacBook Pro to one with the M2 chip.

**Comment by DavidGzzMilan** [comment]
@hemanshukumar I was starting to configure the M1 equipment where I migrated from the Intel one, so I decided to reset it again and skip the migration step. I configured everything manually this time, and I got Homebrew working with no issues :)
Definitely, the problem is somewhere during the migration process from Intel to the M1/M2.

**Comment by hemanshukumar** [comment]
Nice! I don't have that option unfortunately, so I'm still hoping someone here will tell me a solution!

**Comment by MajidiyM** [comment]
I am having this kind of problem while installing homebrew:
Downloading and installing Homebrew...
remote: Enumerating objects: 259256, done.
remote: Counting objects: 100% (31735/31735), done.
remote: Compressing objects: 100% (1061/1061), done.
error: 3606 bytes of body are still expected4 MiB | 187.00 KiB/s
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Failed during: /usr/bin/git fetch --force origin
Maybe someone can help me?

**Comment by diogokings** [comment]
I'm facing the same issue

**Comment by katower** [comment]
I'm facing the same issue

**Comment by delfireinoso** [comment]
You are triing to instal Homebrew from the Git repository?

You have tried the official install process?

https://brew.sh/

**Comment by hamid-kordi** [comment]
I have the same problem in version Sonoma 14.3.1

**Comment by dapperdrop** [comment]
I had the same issue, root cause being similar to several users on this thread. Upgraded/migrated from an x86 MacBook to a arm64 MacBook.

It turned out that I had both x86 and arm64 versions of homebrew installed. To determine if you have the same thing going on:

```
~ which -a brew
/usr/local/bin/brew
/opt/homebrew/bin/brew
/usr/local/bin/brew
```

`/usr/local/bin/brew` = x86 install
`/opt/homebrew/bin/brew` = arm64 install

The solution that worked for me was to uninstall the x86 version only. Steps to do so:

1. Download the uninstall script from: https://github.com/Homebrew/install#uninstall-homebrew
2. Make the script executable: `chmod u+x uninstall.sh`
3. Run the script using a flag to target the x86 install: `sudo ./uninstall.sh --path=/usr/local`
4. Check that you now only have the arm64 version installed: `which -a brew`

Good luck to you all.

**Comment by goodpeter-sun** [comment]
I am facing the same issue. Looking forward to any useful tips here.

**Comment by okineadev** [comment]
```
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```

Doesn't that tell you anything?
